### PR TITLE
Add support for pruning nodes from a DFS.

### PR DESCRIPTION
**Pull request by khuey**
Since you closed my other PR rather than allowing me to push changes to it ...

### CODE REVIEW

File: src/visit/dfsvisit.rs [hunk 1]
```diff
@@ -60,26 +69,38 @@ impl<B> Control<B> {
...
+        }
+    }
 }
 
-impl<E> ControlFlow for Result<(), E> {
-    fn continuing() -> Self { Ok(()) }
+impl<C: ControlFlow, E> ControlFlow for Result<C, E> {
+    fn continuing() -> Self { Ok(C::continuing()) }
     fn should_break(&self) -> bool {
         if let Err(_) = *self { true } else { false }
```
Comment (bluss) [inline src/visit/dfsvisit.rs]: We can pass through the "should_break" here like in should_prune. I think that's a good way to combine the power of Result with the control flow thing.

**Comment by bluss** [comment]
This looks like a good addition, (What do others think?). Ready to go once the last 0.4.x release is out of the way.

**Comment by jmcomets** [comment]
Looks great!

**Comment by bluss** [comment]
It seems like we really only have micro changes unreleased in master, so I guess we can just roll in the changes for 0.5 in this PR and others

**Comment by Dylan-DPC** [comment]
@bluss i think it is better to do it in a seperate release-tracking PR. I can do the release if you want.

**Comment by bluss** [comment]
Ah yeah I don't know why I wrote it like that, I don't mean that we bring other changes into this PR. The 0.4.12 release is done already.. Tracking everything for 0.5 sounds good

**Comment by bluss** [comment]
Thanks for this, and I'm sorry for the delay

**Comment by khuey** [comment]
Thanks for merging this!

**Pull request merged**